cli/flox-rust-sdk/src/models/environment/mod.rs

```diff
@@ -1072,4 +1133,92 @@ mod test {
         environment.lock(&flox).unwrap();
         assert!(environment.needs_migration_to_v1(&flox).unwrap().is_none());
     }
+
+    #[test]
+    fn stable_path_hash() {
+        // Ensure that running the path_hash function gives you the same results
+        // with the same input e.g. doesn't depend on time, etc
+        let (_flox, tmp_dir) = flox_instance();
+        let path = tmp_dir.path().join("foo");
+        std::fs::File::create(&path).unwrap();
+        let path = CanonicalPath::new(path).unwrap();
+
+        let hash1 = path_hash(&path);
+        std::thread::sleep(Duration::from_millis(1_000));
+        let hash2 = path_hash(&path);
+        assert_eq!(hash1, hash2);
+    }
+
+    #[cfg(target_os = "linux")]
+    #[test]
+    fn services_socket_path_respects_xdg_runtime_dir() {
+        let (flox, _temp_dir_handle) = flox_instance();
+        // In reality XDG_RUNTIME_DIR would be something like `/run/user/1001`,
+        // but that won't necessarily exist where this unit test is run.
+        // We need a directory with group and others rights 00 otherwise
+        // xdg::BaseDirectories errors.
+        // And it needs to result in a path shorter than 107 characters.
+        let tempdir = tempfile::Builder::new()
+            .permissions(std::fs::Permissions::from_mode(0o700))
+            .tempdir_in("/tmp")
+            .unwrap();
```

Comment:
This feels a bit awkward and made me wonder if we should keep using xdg::BaseDirectories rather than just using `env::var("XDG_RUNTIME_DIR")`, but I kept using it for consistency